modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/ASCIIFoldingTokenFilterFactory.java

```diff
@@ -68,4 +68,9 @@ public TokenStream create(TokenStream tokenStream) {
             };
         }
     }
+
+    @Override
+    public Object getMultiTermComponent() {
+        return getSynonymFilter(true);
```

Comment:
I wonder if this work since the synonym filter checks only the first token of each position so if the first token is the original one there is no chance that the modified one can match.

Reply:
The folded token is always emitted first, so that's the one we need to match against the synonym map.  Hence, we return a filter that only emits folded tokens.